lib/engine/step/g_1848/dutch_auction.rb

```diff
@@ -0,0 +1,142 @@
+# frozen_string_literal: true
+
+require_relative '../base'
+require_relative '../auctioner'
+
+module Engine
+  module Step
+    module G1848
+      class DutchAuction < Base
+        include Auctioner
+
+        attr_reader :companies
+
+        # On your turn, must buy any private or reduce price of any private
+        # Once you own a private, you may pass
+        # if everything is reduced to minimum price, and nobody has bought a private, you must buy one
+        # if any privates are left and everyone passes in a row, owned privates pay and auction continues
+        # if someone has bought a private and everything is minimum, you can pass
+
+        ACTIONS = %w[bid reduce].freeze
+        ACTIONS_WITH_PASS = %w[bid reduce pass].freeze
+
+        MIN_PRICES = {
+          'P1' => 0,
+          'P2' => 40,
+          'P3' => 80,
+          'P4' => 140,
+          'P5' => 140,
+          'P6' => 200,
+        }.freeze
+
+        def description
+          "Buy a Company or Reduce its Price by #{@game.format_currency(5)}"
+        end
+
+        def available
+          @companies
+        end
+
+        def auctioneer?
+          false
+        end
+
+        def committed_cash(_player, _show_hidden = false)
+          0
+        end
+
+        def pass_description
+          'Pass'
+        end
+
+        def process_bid(action)
+          company = action.company
+          price = company.min_bid
+          buy_company(current_entity, company, price)
+          @round.next_entity_index!
+        end
+
+        def process_pass(action)
+          entity = action.entity
+          @log << "#{entity.name} passes"
+          entity.pass!
+          all_passed! if entities.all?(&:passed?)
+          @round.next_entity_index!
+        end
+
+        def process_reduce(action)
+          action.entity.unpass!
+          company = action.company
+          company.discount += 5
+          price = company.min_bid
+          @log << "#{current_entity.name} reduces #{company.name} by £5 to #{@game.format_currency(price)}"
+          @round.next_entity_index!
+        end
+
+        def may_reduce?(company)
+          # Each private can be discounted a maximum of 6 times
+          company.min_bid > MIN_PRICES[company.sym]
+        end
+
+        def actions(entity)
+          return [] if @companies.empty?
+          return [] unless entity.player?
+
+          actions = entity.player.companies.empty? ? ACTIONS : ACTIONS_WITH_PASS
+
+          entity == current_entity ? actions : []
```

Comment:
why don't you do this current_entity check on line 83

Reply:
I think this was copy/pasted from [the base `waterfall_auction`](https://github.com/tobymao/18xx/blob/master/lib/engine/step/waterfall_auction.rb#L64)

See 9b7e7b30